examples/QIR/Optimization/README.md

```diff
@@ -0,0 +1,274 @@
+# QIR Generation and Optimization
+
+Since QIR is a specification of LLVM IR, the IR code can be manipulated via the usual tools provided by LLVM, such as [Clang](https://clang.llvm.org/), the [LLVM optimizer](https://llvm.org/docs/CommandGuide/opt.html), the [LLVM static compiler](https://llvm.org/docs/CommandGuide/llc.html), and the [just-in-time (JIT) compiler](https://llvm.org/docs/CommandGuide/lli.html).
+This example is structured as a walk-through of the process covering installation, QIR generation, and running built-in and custom LLVM optimization pipelines.
+
+## Prerequisites
+
+QIR generation is handled by the Q# compiler, while optimizations are performed at the LLVM level.
+The following software will be used in this walk-through:
+
+* Quantum Development Kit (QDK), containing the Q# compiler 
+* Clang, an LLVM based compiler for the C language family
+* LLVM optimizer, a tool to run custom optimization pipelines on LLVM IR code
+
+### Installing the QDK
+
+The *Quantum Development Kit (QDK)* documentation provides detailed guides on [installing the QDK](https://docs.microsoft.com/en-us/azure/quantum/install-overview-qdk#install-the-qdk-for-quantum-computing-locally) for various setups.
+This document will assume a command line setup for standalone Q# applications.
+
+Steps:
+
+* Install the [.NET Core SDK 3.1](https://dotnet.microsoft.com/download) (NOTE: use `dotnet-sdk-3.1` instead of `dotnet-sdk-5.0` in the Linux guides)
+* Install the QDK with `dotnet new -i Microsoft.Quantum.ProjectTemplates`
+* (Linux only) Ensure the [GNU OpenMP support library](https://gcc.gnu.org/onlinedocs/libgomp/) is installed on your system, e.g. via `sudo apt install libgomp1`
```

Comment:
It may be good to defer to the install guide for these steps, to make sure that these steps don't get out of date (e.g.: .NET 6 is coming in November).

Reply:
That's a valid point. I added the QDK version for which the instructions are valid and a warning to check the guide for most up-to-date instructions.